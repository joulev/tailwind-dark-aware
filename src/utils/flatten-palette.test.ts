import flattenPalette from "./flatten-palette";
import type { ColourObject } from "$types";

it("Should work for default Tailwind colour palette", () => {
  const flattened = flattenPalette(require("tailwindcss/colors"));
  expect(flattened).toMatchSnapshot();
});

it("Should pass trivial usage", () => {
  const colours: ColourObject = {
    red: { light: "0", dark: "1" },
    blue: { light: "2", dark: "3" },
  };
  const flattened = flattenPalette(colours);
  expect(flattened).toStrictEqual({
    "red-light": ["red", "light"],
    "red-dark": ["red", "dark"],
    "blue-light": ["blue", "light"],
    "blue-dark": ["blue", "dark"],
  });
});

it("Should work with nested colours", () => {
  const colours: ColourObject = {
    red: { light: "0", dark: "1", nested: { light: "2", dark: "3" } },
  };
  const flattened = flattenPalette(colours);
  expect(flattened).toStrictEqual({
    "red-light": ["red", "light"],
    "red-dark": ["red", "dark"],
    "red-nested-light": ["red.nested", "light"],
    "red-nested-dark": ["red.nested", "dark"],
  });
});

it('Should work with "global" colours', () => {
  const colours: ColourObject = {
    primary: "0",
    red: { light: "1", dark: "2" },
  };
  const flattened = flattenPalette(colours);
  expect(flattened).toStrictEqual({
    primary: ["", "primary"],
    "red-light": ["red", "light"],
    "red-dark": ["red", "dark"],
  });
});

it("Should ignore instances of inherit, currentColor, transparent and auto", () => {
  const colours: ColourObject = {
    inherit: "inherit",
    currentColor: "currentColor",
    transparent: "transparent",
    auto: "auto",
    red: {
      light: "0",
      inherit: "inherit",
      currentColor: "currentColor",
      transparent: "transparent",
      auto: "auto",
    },
  };
  const flattened = flattenPalette(colours);
  expect(flattened).toStrictEqual({ "red-light": ["red", "light"] });
});

it("Should ignore colours of key `50`", () => {
  const colours: ColourObject = { red: { light: "0", 50: "1" } };
  const flattened = flattenPalette(colours);
  expect(flattened).toStrictEqual({ "red-light": ["red", "light"] });
});

it("Should handle `DEFAULT` correctly", () => {
  const colours: ColourObject = {
    red: {
      light: "0",
      DEFAULT: "1",
      dark: "2",
      nested: { light: "3", DEFAULT: "4", dark: "5" },
    },
  };
  const flattened = flattenPalette(colours);
  expect(flattened).toStrictEqual({
    "red-light": ["red", "light"],
    red: ["red", "DEFAULT"],
    "red-dark": ["red", "dark"],
    "red-nested-light": ["red.nested", "light"],
    "red-nested": ["red.nested", "DEFAULT"],
    "red-nested-dark": ["red.nested", "dark"],
  });
});

it("Should handle `undefined`", () => {
  const flattened = flattenPalette(undefined);
  expect(flattened).toStrictEqual({});
});

it("Should exclude global DEFAULT colour", () => {
  // if some idiot decides to use the name "DEFAULT" for a global colour...
  const colours: ColourObject = { DEFAULT: "0" };
  const flattened = flattenPalette(colours);
  expect(flattened).toStrictEqual({});
});
