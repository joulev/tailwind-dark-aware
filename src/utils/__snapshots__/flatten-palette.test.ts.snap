// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should work for default Tailwind colour palette 1`] = `
{
  "amber-100": [
    "amber",
    "100",
  ],
  "amber-200": [
    "amber",
    "200",
  ],
  "amber-300": [
    "amber",
    "300",
  ],
  "amber-400": [
    "amber",
    "400",
  ],
  "amber-500": [
    "amber",
    "500",
  ],
  "amber-600": [
    "amber",
    "600",
  ],
  "amber-700": [
    "amber",
    "700",
  ],
  "amber-800": [
    "amber",
    "800",
  ],
  "amber-900": [
    "amber",
    "900",
  ],
  "black": [
    "",
    "black",
  ],
  "blue-100": [
    "blue",
    "100",
  ],
  "blue-200": [
    "blue",
    "200",
  ],
  "blue-300": [
    "blue",
    "300",
  ],
  "blue-400": [
    "blue",
    "400",
  ],
  "blue-500": [
    "blue",
    "500",
  ],
  "blue-600": [
    "blue",
    "600",
  ],
  "blue-700": [
    "blue",
    "700",
  ],
  "blue-800": [
    "blue",
    "800",
  ],
  "blue-900": [
    "blue",
    "900",
  ],
  "blueGray-100": [
    "blueGray",
    "100",
  ],
  "blueGray-200": [
    "blueGray",
    "200",
  ],
  "blueGray-300": [
    "blueGray",
    "300",
  ],
  "blueGray-400": [
    "blueGray",
    "400",
  ],
  "blueGray-500": [
    "blueGray",
    "500",
  ],
  "blueGray-600": [
    "blueGray",
    "600",
  ],
  "blueGray-700": [
    "blueGray",
    "700",
  ],
  "blueGray-800": [
    "blueGray",
    "800",
  ],
  "blueGray-900": [
    "blueGray",
    "900",
  ],
  "coolGray-100": [
    "coolGray",
    "100",
  ],
  "coolGray-200": [
    "coolGray",
    "200",
  ],
  "coolGray-300": [
    "coolGray",
    "300",
  ],
  "coolGray-400": [
    "coolGray",
    "400",
  ],
  "coolGray-500": [
    "coolGray",
    "500",
  ],
  "coolGray-600": [
    "coolGray",
    "600",
  ],
  "coolGray-700": [
    "coolGray",
    "700",
  ],
  "coolGray-800": [
    "coolGray",
    "800",
  ],
  "coolGray-900": [
    "coolGray",
    "900",
  ],
  "cyan-100": [
    "cyan",
    "100",
  ],
  "cyan-200": [
    "cyan",
    "200",
  ],
  "cyan-300": [
    "cyan",
    "300",
  ],
  "cyan-400": [
    "cyan",
    "400",
  ],
  "cyan-500": [
    "cyan",
    "500",
  ],
  "cyan-600": [
    "cyan",
    "600",
  ],
  "cyan-700": [
    "cyan",
    "700",
  ],
  "cyan-800": [
    "cyan",
    "800",
  ],
  "cyan-900": [
    "cyan",
    "900",
  ],
  "emerald-100": [
    "emerald",
    "100",
  ],
  "emerald-200": [
    "emerald",
    "200",
  ],
  "emerald-300": [
    "emerald",
    "300",
  ],
  "emerald-400": [
    "emerald",
    "400",
  ],
  "emerald-500": [
    "emerald",
    "500",
  ],
  "emerald-600": [
    "emerald",
    "600",
  ],
  "emerald-700": [
    "emerald",
    "700",
  ],
  "emerald-800": [
    "emerald",
    "800",
  ],
  "emerald-900": [
    "emerald",
    "900",
  ],
  "fuchsia-100": [
    "fuchsia",
    "100",
  ],
  "fuchsia-200": [
    "fuchsia",
    "200",
  ],
  "fuchsia-300": [
    "fuchsia",
    "300",
  ],
  "fuchsia-400": [
    "fuchsia",
    "400",
  ],
  "fuchsia-500": [
    "fuchsia",
    "500",
  ],
  "fuchsia-600": [
    "fuchsia",
    "600",
  ],
  "fuchsia-700": [
    "fuchsia",
    "700",
  ],
  "fuchsia-800": [
    "fuchsia",
    "800",
  ],
  "fuchsia-900": [
    "fuchsia",
    "900",
  ],
  "gray-100": [
    "gray",
    "100",
  ],
  "gray-200": [
    "gray",
    "200",
  ],
  "gray-300": [
    "gray",
    "300",
  ],
  "gray-400": [
    "gray",
    "400",
  ],
  "gray-500": [
    "gray",
    "500",
  ],
  "gray-600": [
    "gray",
    "600",
  ],
  "gray-700": [
    "gray",
    "700",
  ],
  "gray-800": [
    "gray",
    "800",
  ],
  "gray-900": [
    "gray",
    "900",
  ],
  "green-100": [
    "green",
    "100",
  ],
  "green-200": [
    "green",
    "200",
  ],
  "green-300": [
    "green",
    "300",
  ],
  "green-400": [
    "green",
    "400",
  ],
  "green-500": [
    "green",
    "500",
  ],
  "green-600": [
    "green",
    "600",
  ],
  "green-700": [
    "green",
    "700",
  ],
  "green-800": [
    "green",
    "800",
  ],
  "green-900": [
    "green",
    "900",
  ],
  "indigo-100": [
    "indigo",
    "100",
  ],
  "indigo-200": [
    "indigo",
    "200",
  ],
  "indigo-300": [
    "indigo",
    "300",
  ],
  "indigo-400": [
    "indigo",
    "400",
  ],
  "indigo-500": [
    "indigo",
    "500",
  ],
  "indigo-600": [
    "indigo",
    "600",
  ],
  "indigo-700": [
    "indigo",
    "700",
  ],
  "indigo-800": [
    "indigo",
    "800",
  ],
  "indigo-900": [
    "indigo",
    "900",
  ],
  "lightBlue-100": [
    "lightBlue",
    "100",
  ],
  "lightBlue-200": [
    "lightBlue",
    "200",
  ],
  "lightBlue-300": [
    "lightBlue",
    "300",
  ],
  "lightBlue-400": [
    "lightBlue",
    "400",
  ],
  "lightBlue-500": [
    "lightBlue",
    "500",
  ],
  "lightBlue-600": [
    "lightBlue",
    "600",
  ],
  "lightBlue-700": [
    "lightBlue",
    "700",
  ],
  "lightBlue-800": [
    "lightBlue",
    "800",
  ],
  "lightBlue-900": [
    "lightBlue",
    "900",
  ],
  "lime-100": [
    "lime",
    "100",
  ],
  "lime-200": [
    "lime",
    "200",
  ],
  "lime-300": [
    "lime",
    "300",
  ],
  "lime-400": [
    "lime",
    "400",
  ],
  "lime-500": [
    "lime",
    "500",
  ],
  "lime-600": [
    "lime",
    "600",
  ],
  "lime-700": [
    "lime",
    "700",
  ],
  "lime-800": [
    "lime",
    "800",
  ],
  "lime-900": [
    "lime",
    "900",
  ],
  "neutral-100": [
    "neutral",
    "100",
  ],
  "neutral-200": [
    "neutral",
    "200",
  ],
  "neutral-300": [
    "neutral",
    "300",
  ],
  "neutral-400": [
    "neutral",
    "400",
  ],
  "neutral-500": [
    "neutral",
    "500",
  ],
  "neutral-600": [
    "neutral",
    "600",
  ],
  "neutral-700": [
    "neutral",
    "700",
  ],
  "neutral-800": [
    "neutral",
    "800",
  ],
  "neutral-900": [
    "neutral",
    "900",
  ],
  "orange-100": [
    "orange",
    "100",
  ],
  "orange-200": [
    "orange",
    "200",
  ],
  "orange-300": [
    "orange",
    "300",
  ],
  "orange-400": [
    "orange",
    "400",
  ],
  "orange-500": [
    "orange",
    "500",
  ],
  "orange-600": [
    "orange",
    "600",
  ],
  "orange-700": [
    "orange",
    "700",
  ],
  "orange-800": [
    "orange",
    "800",
  ],
  "orange-900": [
    "orange",
    "900",
  ],
  "pink-100": [
    "pink",
    "100",
  ],
  "pink-200": [
    "pink",
    "200",
  ],
  "pink-300": [
    "pink",
    "300",
  ],
  "pink-400": [
    "pink",
    "400",
  ],
  "pink-500": [
    "pink",
    "500",
  ],
  "pink-600": [
    "pink",
    "600",
  ],
  "pink-700": [
    "pink",
    "700",
  ],
  "pink-800": [
    "pink",
    "800",
  ],
  "pink-900": [
    "pink",
    "900",
  ],
  "purple-100": [
    "purple",
    "100",
  ],
  "purple-200": [
    "purple",
    "200",
  ],
  "purple-300": [
    "purple",
    "300",
  ],
  "purple-400": [
    "purple",
    "400",
  ],
  "purple-500": [
    "purple",
    "500",
  ],
  "purple-600": [
    "purple",
    "600",
  ],
  "purple-700": [
    "purple",
    "700",
  ],
  "purple-800": [
    "purple",
    "800",
  ],
  "purple-900": [
    "purple",
    "900",
  ],
  "red-100": [
    "red",
    "100",
  ],
  "red-200": [
    "red",
    "200",
  ],
  "red-300": [
    "red",
    "300",
  ],
  "red-400": [
    "red",
    "400",
  ],
  "red-500": [
    "red",
    "500",
  ],
  "red-600": [
    "red",
    "600",
  ],
  "red-700": [
    "red",
    "700",
  ],
  "red-800": [
    "red",
    "800",
  ],
  "red-900": [
    "red",
    "900",
  ],
  "rose-100": [
    "rose",
    "100",
  ],
  "rose-200": [
    "rose",
    "200",
  ],
  "rose-300": [
    "rose",
    "300",
  ],
  "rose-400": [
    "rose",
    "400",
  ],
  "rose-500": [
    "rose",
    "500",
  ],
  "rose-600": [
    "rose",
    "600",
  ],
  "rose-700": [
    "rose",
    "700",
  ],
  "rose-800": [
    "rose",
    "800",
  ],
  "rose-900": [
    "rose",
    "900",
  ],
  "sky-100": [
    "sky",
    "100",
  ],
  "sky-200": [
    "sky",
    "200",
  ],
  "sky-300": [
    "sky",
    "300",
  ],
  "sky-400": [
    "sky",
    "400",
  ],
  "sky-500": [
    "sky",
    "500",
  ],
  "sky-600": [
    "sky",
    "600",
  ],
  "sky-700": [
    "sky",
    "700",
  ],
  "sky-800": [
    "sky",
    "800",
  ],
  "sky-900": [
    "sky",
    "900",
  ],
  "slate-100": [
    "slate",
    "100",
  ],
  "slate-200": [
    "slate",
    "200",
  ],
  "slate-300": [
    "slate",
    "300",
  ],
  "slate-400": [
    "slate",
    "400",
  ],
  "slate-500": [
    "slate",
    "500",
  ],
  "slate-600": [
    "slate",
    "600",
  ],
  "slate-700": [
    "slate",
    "700",
  ],
  "slate-800": [
    "slate",
    "800",
  ],
  "slate-900": [
    "slate",
    "900",
  ],
  "stone-100": [
    "stone",
    "100",
  ],
  "stone-200": [
    "stone",
    "200",
  ],
  "stone-300": [
    "stone",
    "300",
  ],
  "stone-400": [
    "stone",
    "400",
  ],
  "stone-500": [
    "stone",
    "500",
  ],
  "stone-600": [
    "stone",
    "600",
  ],
  "stone-700": [
    "stone",
    "700",
  ],
  "stone-800": [
    "stone",
    "800",
  ],
  "stone-900": [
    "stone",
    "900",
  ],
  "teal-100": [
    "teal",
    "100",
  ],
  "teal-200": [
    "teal",
    "200",
  ],
  "teal-300": [
    "teal",
    "300",
  ],
  "teal-400": [
    "teal",
    "400",
  ],
  "teal-500": [
    "teal",
    "500",
  ],
  "teal-600": [
    "teal",
    "600",
  ],
  "teal-700": [
    "teal",
    "700",
  ],
  "teal-800": [
    "teal",
    "800",
  ],
  "teal-900": [
    "teal",
    "900",
  ],
  "trueGray-100": [
    "trueGray",
    "100",
  ],
  "trueGray-200": [
    "trueGray",
    "200",
  ],
  "trueGray-300": [
    "trueGray",
    "300",
  ],
  "trueGray-400": [
    "trueGray",
    "400",
  ],
  "trueGray-500": [
    "trueGray",
    "500",
  ],
  "trueGray-600": [
    "trueGray",
    "600",
  ],
  "trueGray-700": [
    "trueGray",
    "700",
  ],
  "trueGray-800": [
    "trueGray",
    "800",
  ],
  "trueGray-900": [
    "trueGray",
    "900",
  ],
  "violet-100": [
    "violet",
    "100",
  ],
  "violet-200": [
    "violet",
    "200",
  ],
  "violet-300": [
    "violet",
    "300",
  ],
  "violet-400": [
    "violet",
    "400",
  ],
  "violet-500": [
    "violet",
    "500",
  ],
  "violet-600": [
    "violet",
    "600",
  ],
  "violet-700": [
    "violet",
    "700",
  ],
  "violet-800": [
    "violet",
    "800",
  ],
  "violet-900": [
    "violet",
    "900",
  ],
  "warmGray-100": [
    "warmGray",
    "100",
  ],
  "warmGray-200": [
    "warmGray",
    "200",
  ],
  "warmGray-300": [
    "warmGray",
    "300",
  ],
  "warmGray-400": [
    "warmGray",
    "400",
  ],
  "warmGray-500": [
    "warmGray",
    "500",
  ],
  "warmGray-600": [
    "warmGray",
    "600",
  ],
  "warmGray-700": [
    "warmGray",
    "700",
  ],
  "warmGray-800": [
    "warmGray",
    "800",
  ],
  "warmGray-900": [
    "warmGray",
    "900",
  ],
  "white": [
    "",
    "white",
  ],
  "yellow-100": [
    "yellow",
    "100",
  ],
  "yellow-200": [
    "yellow",
    "200",
  ],
  "yellow-300": [
    "yellow",
    "300",
  ],
  "yellow-400": [
    "yellow",
    "400",
  ],
  "yellow-500": [
    "yellow",
    "500",
  ],
  "yellow-600": [
    "yellow",
    "600",
  ],
  "yellow-700": [
    "yellow",
    "700",
  ],
  "yellow-800": [
    "yellow",
    "800",
  ],
  "yellow-900": [
    "yellow",
    "900",
  ],
  "zinc-100": [
    "zinc",
    "100",
  ],
  "zinc-200": [
    "zinc",
    "200",
  ],
  "zinc-300": [
    "zinc",
    "300",
  ],
  "zinc-400": [
    "zinc",
    "400",
  ],
  "zinc-500": [
    "zinc",
    "500",
  ],
  "zinc-600": [
    "zinc",
    "600",
  ],
  "zinc-700": [
    "zinc",
    "700",
  ],
  "zinc-800": [
    "zinc",
    "800",
  ],
  "zinc-900": [
    "zinc",
    "900",
  ],
}
`;
